Userland/Libraries/LibWeb/PixelUnits.h

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2022, Sam Atkins <atkinssj@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <AK/DistinctNumeric.h>
+#include <AK/Traits.h>
+#include <LibGfx/Forward.h>
+#include <math.h>
+
+namespace Web {
+
+/// DevicePixels: A position or length on the physical display.
+class DevicePixels {
```

Comment:
Isn't this just basically `using DevicePixels = float`?
I'm sure it's not

Reply:
I believe `using DevicePixels = int` would just be an alias for `int`. Making it a class forces you to explicitly convert to/from `DevivePixels` (which I believe is the intention here).

Though it does seem annoying you have to write out this much boilerplate for each class.

Reply:
That's what `DistinctNumeric` is for

Reply:
You could maybe half the boilerplate if both CSSPixels and DevicePixels inherited from some template base will all the operators. `class CSSPixels: public ArithmeticOperations<CSSPixels> {` (maybe).

Reply:
I tried `DistinctNumeric` originally, but I'd wanted to keep one of the constructor private to restrict who could convert one type to the other. I ended up not doing that though.

I'll give it another try. Would be nice to not have all this boilerplate - I already had one silly typo in it that confused me for ages.

Reply:
Oh, lol, I'm even still including `<AK/DistinctNumeric.h>` :laughing: